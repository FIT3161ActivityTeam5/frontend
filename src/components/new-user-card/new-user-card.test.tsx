import React from 'react';
import renderer from 'react-test-renderer';
import AuthenticationContext from '../../contexts/authentication-context';

import NewUserCard from './new-user-card';

describe('<NewUserCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <AuthenticationContext.Provider value={{
        accessToken: 'abc',
        isLoading: false,
        login: jest.fn(),
        logout: jest.fn(),
      }}>
        <NewUserCard onCreateFirst={jest.fn()} />
      </AuthenticationContext.Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
