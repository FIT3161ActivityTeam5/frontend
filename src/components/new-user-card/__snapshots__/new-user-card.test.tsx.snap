// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewUserCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingBottom": 32,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 32,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_400Regular",
        },
        Object {
          "color": "rgba(75, 85, 99, 1)",
          "fontSize": 20,
          "lineHeight": 28,
          "textAlign": "center",
        },
      ]
    }
  >
    Looks like you don't have any maps, press the button below to get started.
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(124, 58, 237, 1)",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "marginTop": 32,
        "opacity": 1,
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 12,
        "width": "66.666667%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins_700Bold",
          },
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
    >
      New Map
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_400Regular",
        },
        Object {
          "color": "rgba(75, 85, 99, 1)",
          "fontSize": 14,
          "lineHeight": 20,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Unsure where to begin?
    

    Check out the user guide in the 
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins_400Regular",
          },
          Object {
            "color": "rgba(109, 40, 217, 1)",
          },
        ]
      }
    >
      settings screen
    </Text>
    .
  </Text>
</View>
`;
